File: Atividade 2 - implementação da lista encadeada/main.py
import time

def executar_operacoes():
    with open('arq-novo.txt', 'r') as arquivo:
        linhas = [linha.strip() for linha in arquivo.readlines()]

    lista = list(map(int, linhas[0].split()))
    qtd_acoes = int(linhas[1])

    for i in range(2, 2 + qtd_acoes):
        partes = linhas[i].split()
        if not partes:
            continue

        acao = partes[0]

        if acao == 'A':  # Adicionar
            numero = int(partes[1])
            posicao = int(partes[2])
            if 0 <= posicao <= len(lista):
                lista.insert(posicao, numero)

        elif acao == 'R':  # Remover
            numero = int(partes[1])
            if numero in lista:
                lista.remove(numero)

        elif acao == 'P':  # Imprimir
            print(' '.join(map(str, lista)))

    return True

# Executar 10 vezes com medição de tempo
for i in range(1, 11):
    print(f"\nExecução {i}:")
    inicio = time.time()
    executar_operacoes()
    fim = time.time()
    tempo_total = fim - inicio
    print(f"Tempo total de execução: {tempo_total:.4f} segundos")

File: Atividade 2 - implementação da lista encadeada/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#define MAX 1000

int lista[MAX];
int tamanho = 0;

void adicionar(int numero, int posicao) {
    if (posicao < 0 || posicao > tamanho || tamanho >= MAX) return;
    for (int i = tamanho; i > posicao; i--) {
        lista[i] = lista[i - 1];
    }
    lista[posicao] = numero;
    tamanho++;
}

void remover(int numero) {
    int i;
    for (i = 0; i < tamanho; i++) {
        if (lista[i] == numero) {
            break;
        }
    }
    if (i == tamanho) return;
    for (; i < tamanho - 1; i++) {
        lista[i] = lista[i + 1];
    }
    tamanho--;
}

void imprimir() {
    for (int i = 0; i < tamanho; i++) {
        printf("%d ", lista[i]);
    }
    printf("\n");
}

void executarOperacoes() {
    FILE *fp = fopen("arq-novo.txt", "r");
    if (!fp) {
        printf("Erro ao abrir o arquivo.\n");
        exit(1);
    }

    // Lê a primeira linha (lista inicial)
    char linha[256];
    fgets(linha, sizeof(linha), fp);
    char *token = strtok(linha, " ");
    tamanho = 0;
    while (token) {
        lista[tamanho++] = atoi(token);
        token = strtok(NULL, " ");
    }

    // Lê a quantidade de ações
    int qtdAcoes;
    fscanf(fp, "%d\n", &qtdAcoes);

    for (int i = 0; i < qtdAcoes; i++) {
        fgets(linha, sizeof(linha), fp);
        char acao;
        int num, pos;
        if (linha[0] == 'A') {
            sscanf(linha, "%c %d %d", &acao, &num, &pos);
            adicionar(num, pos);
        } else if (linha[0] == 'R') {
            sscanf(linha, "%c %d", &acao, &num);
            remover(num);
        } else if (linha[0] == 'P') {
            imprimir();
        }
    }

    fclose(fp);
}

int main() {
    for (int i = 1; i <= 10; i++) {
        printf("\nExecução %d:\n", i);
        clock_t inicio = clock();
        executarOperacoes();
        clock_t fim = clock();
        double tempo = (double)(fim - inicio) / CLOCKS_PER_SEC;
        printf("Tempo total de execução: %.4f segundos\n", tempo);
    }
    return 0;
}
